assets/js/base/hooks/use-store-cart-item.js

```diff
@@ -22,32 +24,43 @@ import { useStoreCart } from './use-store-cart';
  */
 export const useStoreCartItem = ( cartItemKey ) => {
 	const { cartItems, cartIsLoading } = useStoreCart();
-	const cartItem = cartItems.filter( ( item ) => item.key === cartItemKey );
+	const cartItem = cartItems.find( ( item ) => item.key === cartItemKey );
```

Comment:
I just realized, deriving the `cartItem` should probably be done in an effect because cartItems might change (incidentally, this is what can cause the issue with the quantity state).
```suggestion
    // note for example I'm just doing `{ quantity: 0, key: '' }` here, 
    // but this should actually be a default object in the shape of
    //  a cartItem schema
    const [ cartItem, setCartItem ] = useState( { quantity: 0, key: '' } );
    useEffect( () => {
        if ( ! cartIsLoading && cartItem.key !== cartItemKey ) {
            	const cartItem = cartItems.find( ( item ) => item.key === cartItemKey );
            	if ( cartItem ) {
            	    setCartItem( cartItem );
            	}
        }
    }, [ cartItems, cartIsLoading, cartItem );
```

Reply:
I tried to get this working, but the default quantity values in the state `cartItem` ended up resetting the real cart item quantity to zero on load, and persistence wasn't working at all.

I also tried a variation where the hook only provides `quantity` state (instead of the full `cartItem`). It didn't feel right having the full cart object in state, especially since the client component doesn't actually use it. However that had the same issue - the default quantity overwrote the real quantity.

So - @nerrad I'm thinking we merge this PR as is since it's testing and working well. We can still of course iterate on how this hook works, but I'm hesitant to get bogged down in this. Or if you want to jump on the PR and implement this please go ahead :)

Note to be clear the optional chain operator is only needed in the editor. I think what's happening is the preview data (no real/ajax data from hooks) is used, and then in `CartLineItemRow` the hook fires which requests real cart items. I see this as a bug – we should prevent this from making any real cart AJAX requests in the editor, that's the real cause of the issue IMO.

Reply:
I accepted the invitation to address this and I think what I've pushed should work.  My bad because yes the original suggested code was not quite right.

See if this works well for you and if it does, let's merge.